crates/subspace-farmer/src/commitments.rs

```diff
@@ -423,31 +421,30 @@ impl Commitments {
         let commitment_clone = self.clone();
         tokio::spawn(async move {
             loop {
-                let guard = commitment_clone.inner.commitment_databases.lock().await;
-                let status = guard.metadata_cache.get(&salt);
-                if status.is_none() {
-                    info!(
-                        "Could not retrieve the DB with salt: {:?}, will try again VERY soon...",
-                        salt
-                    );
-                    drop(guard);
-                    sleep(Duration::from_millis(100)).await;
-                    continue;
-                }
-                info!("Successfully retrieved the DB with salt: {:?}", salt);
-                match status.unwrap() {
-                    CommitmentStatus::InProgress => {
-                        // drop the guard, so commitment can make progress
+                if let Some(guard) = commitment_clone.inner.commitment_databases.try_lock() {
+                    let status = guard.metadata_cache.get(&salt);
+                    if status.is_none() {
                         drop(guard);
                         sleep(Duration::from_millis(100)).await;
```

Comment:
This method should be completely different. It should allow to subscribe to an event instead of taking locks on databases and having sleeps. This implementation is basically inverse from what it should be, so I would prefer to just change it to what it should be instead of iterating in this direction.

Reply:
I'll try my best :)